graylog2-server/src/main/java/org/graylog2/bootstrap/preflight/PreflightConfig.java

```diff
@@ -17,15 +17,16 @@
 package org.graylog2.bootstrap.preflight;
 
 import org.bson.types.ObjectId;
-import org.graylog2.database.CollectionName;
+import org.graylog2.database.DbEntity;
 import org.graylog2.database.PersistedImpl;
 import org.graylog2.plugin.database.validators.Validator;
 
 import javax.annotation.Nullable;
 import java.util.Collections;
 import java.util.Map;
+import java.util.Optional;
 
-@CollectionName("preflight")
+@DbEntity(collection = "preflight")
```

Comment:
`PreflightConfig` documents do not seem to have `title` field.
You should probably override default `titleField` property of `DBEntity` annotation.

`PreflightConfig` does not seem to be hidden behind any permissions.
Maybe you should consider changing default `readPermission` property from `NOBODY_ALOWED` to `ALL_ALOWED`?